### ISSUE TITLE
Transcoding PCM audio to Wav

### ISSUE THREAD

**Issue by AlexSapple** [rosenbjerg/FFMpegCore#112]
Hi guys, 

quick question. If I had a network stream of audio, and I wanted to transcode for the purpose of turning a PCM audio in to a wav audio, can I do something similar to the following;

    //should be 16 bits per sample with a 16000 sampling rate, single channel.
    await FFMpegArguments
    .FromPipe(new StreamPipeSource(
        await _audioClient.GetStreamAsync(Id)))
    .WithAudioCodec("u16le")
    .CopyChannel(Channel.Audio)
    .WithAudioSamplingRate(16000)
    .WithAudioBitrate(16)
    .OutputToFile("AlexSappleAudioFeed.wav")
    .ProcessAsynchronously();

unfortunately - with this all I have got so far is "Output file not created". My guess is the codec maybe incorrect (but I'm quite new to this sort of thing). 

couple of questions; 

1. Can I see the underlying FFMpeg command that FFMpegSharp actually runs? 
2. Is there any more verbose debugging options/output. 

I'm also potentially quite keen to output to a pipe too - but 1 step at a time :) 

thanks

**Comment by rosenbjerg** [comment]
You could try to use .ForceFormat to specify the input codec, right after the FromPipe

**Comment by AlexSapple** [comment]
Hi @rosenbjerg 

thanks for your reply. Do you mean use ForceFormat instead of WithAudioCodec? 
so .ForceFormat("s16le") ? 

please forgive me if I'm not making a lot of sense, working with audio and video is a very new subject for me. I gather the error "output file not created" is as good as I'm going to get in terms of detail on why it's failing. 

I've done something relatively similar with H264 video frames to create a hls container and that worked, so I can't help but feel I'm close... :)

**Comment by AlexSapple** [comment]
Hi @rosenbjerg 

The following FFMPeg command works on my audio file

ffmpeg -f s16le -ar 16k -ac 1 -i audiofeed.pcm file.wav

but whatever I try to get this to work via FFMpeg core seems to just gives me the error. Would you or anyone else here be able to see how I can effectively get the wrapper to create a command similar to the above?

**Comment by AlexSapple** [comment]
I've revisited this a few times over the last few days. I still cannot make it write the file. The exception does not contain any FFMpegErrorOutput but the FFMpegExceptionType type is "File".

I have considered whether my code can write to the location and whether the input file exists and output directory exists etc and all seems fine. 

I've tried simplifying the command as much as possible but always the same error. 

    await FFMpegArguments
       .FromInputFiles("audiofeed.pcm")
       .ForceFormat("s16le -ar 16k -ac 1 ")
       .WithAudioCodec(AudioCodec.Aac)
       .OutputToFile("output.aac")
       .ProcessAsynchronously();


I'm expecting this to build something along the lines of (which works when run directly through command line);

    ffmpeg -f s16le -ar 16k -ac 1 -i audiofeed.pcm -c:a aac output.aac

Would really appreciate the help! (it's starting to drive me mad)
